src/coreclr/vm/interpexec.cpp

```diff
@@ -48,16 +52,714 @@ void InterpExecMethod(InterpMethodContextFrame *pFrame, InterpThreadContext *pTh
+                ip += 4;
+                break;
+            case INTOP_CLT_I8:
+                LOCAL_VAR(ip[1], int32_t) = LOCAL_VAR(ip[2], int64_t) < LOCAL_VAR(ip[3], int64_t);
+                ip += 4;
+                break;
+            case INTOP_CLT_R4:
+                CMP_BINOP_FP(float, <, 0);
+                break;
+            case INTOP_CLT_R8:
+                CMP_BINOP_FP(double, <, 0);
+                break;
+
+            case INTOP_CLT_UN_I4:
+                LOCAL_VAR(ip[1], int32_t) = LOCAL_VAR(ip[2], uint32_t) < LOCAL_VAR(ip[3], uint32_t);
+                ip += 4;
+                break;
+            case INTOP_CLT_UN_I8:
+                LOCAL_VAR(ip[1], int32_t) = LOCAL_VAR(ip[2], uint64_t) < LOCAL_VAR(ip[3], uint64_t);
+                ip += 4;
+                break;
+            case INTOP_CLT_UN_R4:
+                CMP_BINOP_FP(float, <, 1);
+                break;
+            case INTOP_CLT_UN_R8:
+                CMP_BINOP_FP(double, <, 1);
+                break;
+
+            case INTOP_CALL:
+            {
+                size_t targetMethod = (size_t)pMethod->pDataItems[ip[3]];
+                returnOffset = ip[1];
+                callArgsOffset = ip[2];
+                const int32_t *targetIp;
+
+                if (targetMethod & INTERP_METHOD_DESC_TAG)
+                {
+                    // First execution of this call. Ensure target method is compiled and
+                    // patch the data item slot with the actual method code.
+                    MethodDesc *pMD = (MethodDesc*)(targetMethod & ~INTERP_METHOD_DESC_TAG);
+                    PCODE code = pMD->GetNativeCode();
+                    if (!code) {
+                        pMD->PrepareInitialCode(CallerGCMode::Coop);
+                        code = pMD->GetNativeCode();
+                    }
+                    pMethod->pDataItems[ip[3]] = (void*)code;
+                    targetIp = (const int32_t*)code;
+                }
+                else
+                {
+                    // At this stage in the implementation, we assume this is pointer to
+                    // interpreter code. In the future, this should probably be tagged pointer
+                    // for interpreter call or normal pointer for JIT/R2R call.
+                    targetIp = (const int32_t*)targetMethod;
+                }
+
+                // Save current execution state once we return from called method
+                pFrame->ip = ip + 4;
+
+                // Allocate child frame.
+                {
+                    InterpMethodContextFrame *pChildFrame = pFrame->pNextFree;
+                    if (!pChildFrame)
+                    {
+                        pChildFrame = (InterpMethodContextFrame*)alloca(sizeof(InterpMethodContextFrame));
+                        pChildFrame->pNextFree = NULL;
+                        // Not free currently, but will be when allocation attempted.
```

Comment:
I don't understand this comment

Reply:
Yes, it is somewhat poorly worded, also the naming of pNextFree can be confusing. The frames are linked with each other backwards (via the parent pointer) but also forward via this `pNextFree` pointer, which should be just renamed to `pNext`.